pkg/strategy/supertrend/strategy.go

```diff
@@ -478,14 +494,59 @@ func (s *Strategy) Run(ctx context.Context, orderExecutor bbgo.OrderExecutor, se
 
 			s.AccumulatedProfitReport.RecordProfit(profit.Profit)
 		})
-		s.orderExecutor.TradeCollector().OnTrade(func(trade types.Trade, profit fixedpoint.Value, netProfit fixedpoint.Value) {
-			s.AccumulatedProfitReport.RecordTrade(trade.Fee)
-		})
+		// s.orderExecutor.TradeCollector().OnTrade(func(trade types.Trade, profit fixedpoint.Value, netProfit fixedpoint.Value) {
+		// 	s.AccumulatedProfitReport.RecordTrade(trade.Fee)
+		// })
 		session.MarketDataStream.OnKLineClosed(types.KLineWith(s.Symbol, types.Interval1d, func(kline types.KLine) {
 			s.AccumulatedProfitReport.DailyUpdate(s.TradeStats)
 		}))
 	}
 
+	// For drawing
+	profitSlice := floats.Slice{1., 1.}
+	price, _ := session.LastPrice(s.Symbol)
+	initAsset := s.CalcAssetValue(price).Float64()
+	cumProfitSlice := floats.Slice{initAsset, initAsset}
+
+	s.orderExecutor.TradeCollector().OnTrade(func(trade types.Trade, profit fixedpoint.Value, netProfit fixedpoint.Value) {
+		if bbgo.IsBackTesting {
+			s.AccumulatedProfitReport.RecordTrade(trade.Fee)
+		}
+
+		// For drawing/charting
+		price := trade.Price.Float64()
+		if s.buyPrice > 0 {
+			profitSlice.Update(price / s.buyPrice)
+			cumProfitSlice.Update(s.CalcAssetValue(trade.Price).Float64())
+		} else if s.sellPrice > 0 {
+			profitSlice.Update(s.sellPrice / price)
+			cumProfitSlice.Update(s.CalcAssetValue(trade.Price).Float64())
+		}
+		if s.Position.IsDust(trade.Price) {
+			s.buyPrice = 0
+			s.sellPrice = 0
+			s.highestPrice = 0
+			s.lowestPrice = 0
+		} else if s.Position.IsLong() {
+			s.buyPrice = price
+			s.sellPrice = 0
+			s.highestPrice = s.buyPrice
+			s.lowestPrice = 0
+		} else {
+			s.sellPrice = price
+			s.buyPrice = 0
+			s.highestPrice = 0
+			s.lowestPrice = s.sellPrice
+		}
+	})
+
+	// event trigger order: s.Interval => Interval1m
+	store, ok := session.SerialMarketDataStore(s.Symbol, []types.Interval{s.Interval, types.Interval1m})
+	if !ok {
+		panic("cannot get 1m history")
```

Comment:
since it's the run method, you can simply return error, by `errors.New` or `fmt.Errorf`

Reply:
Done